File: lib/models/item_model.dart
class ItemModel {
  final int realIndex;
  String no;
  String displayNo; // ❗ 화면 표시용 가상 번호
  String itemCode;
  String quantity;
  bool complete;
  String complement; 
  String process;    
  String remarks;
  String processTime;      // ❗ 공정 기록 시간
  String complementTime;   // ❗ 보완 기록 시간
  bool isSubheading;
  String subheadingTitle; // ❗ 해당 항목이 속한 부분제목

  ItemModel({
    required this.realIndex,
    required this.no,
    required this.displayNo,
    required this.itemCode,
    required this.quantity,
    this.complete = false,
    this.complement = "",
    this.process = "",
    this.remarks = "",
    this.processTime = "",
    this.complementTime = "",
    this.isSubheading = false,
    this.subheadingTitle = "",
  }) {
    // ❗ "null" 문자열 방어 코드 (생성 시점에 빈값으로 치환)
    if (this.no.toLowerCase() == "null") this.no = "";
    if (this.displayNo.toLowerCase() == "null") this.displayNo = "";
    if (this.itemCode.toLowerCase() == "null") this.itemCode = "";
    if (this.quantity.toLowerCase() == "null") this.quantity = "";
    if (this.complement.toLowerCase() == "null") this.complement = "";
    if (this.process.toLowerCase() == "null") this.process = "";
    if (this.remarks.toLowerCase() == "null") this.remarks = "";
    if (this.processTime.toLowerCase() == "null") this.processTime = "";
    if (this.complementTime.toLowerCase() == "null") this.complementTime = "";
    if (this.subheadingTitle.toLowerCase() == "null") this.subheadingTitle = "";
  }
}

File: android/app/src/main/kotlin/org/example/checksheet/models/ItemModel.kt
package org.example.checksheet.models

data class ItemModel(
    val realIndex: Int,
    var no: String,
    var displayNo: String,
    var itemCode: String,
    var quantity: String,
    var complete: Boolean = false,
    var complement: String = "",
    var process: String = "",
    var remarks: String = "",
    var processTime: String = "",
    var complementTime: String = "",
    var completeTime: String = "",
    var isSubheading: Boolean = false,
    var subheadingTitle: String = ""
) {
    init {
        if (no.lowercase() == "null") no = ""
        if (displayNo.lowercase() == "null") displayNo = ""
        if (itemCode.lowercase() == "null") itemCode = ""
        if (quantity.lowercase() == "null") quantity = ""
        if (complement.lowercase() == "null") complement = ""
        if (process.lowercase() == "null") process = ""
        if (remarks.lowercase() == "null") remarks = ""
        if (processTime.lowercase() == "null") processTime = ""
        if (complementTime.lowercase() == "null") complementTime = ""
        if (completeTime.lowercase() == "null") completeTime = ""
        if (subheadingTitle.lowercase() == "null") subheadingTitle = ""
    }
}
